src/ngraph/pass/algebraic_simplification.cpp

```diff
@@ -543,6 +546,77 @@ static bool simplify_reduction(shared_ptr<Node> n)
     return true;
 }
 
+// input({1, ?}) -> transpose({1, 0}) -> reshape() => input({1,?}) -> reshape()
+// input -> transpose({1, 0}) -> squeeze({1}) => {1, 0} -> gather({1}) -> squeeze()
+static bool simplify_transpose(shared_ptr<Node> n)
+{
+    NGRAPH_DEBUG << "In simplify_transpose for " << n->get_name();
+    auto squeeze = as_type_ptr<op::Squeeze>(n);
+    auto reshape = as_type_ptr<op::Reshape>(n);
+
+    // create transpose squeeze pattern
+    auto input = make_shared<pattern::op::Label>(element::i64, Shape{1, 2});
+    auto cnst_perm_lable =
+        make_shared<pattern::op::Label>(element::i64, Shape{2}, pattern::has_class<op::Constant>());
+    auto t = make_shared<op::Transpose>(input, cnst_perm_lable);
+
+    if (squeeze)
+    {
+        auto cnst_axis_lable = make_shared<pattern::op::Label>(
+            element::i64, Shape{1}, pattern::has_class<op::Constant>());
+        auto s = make_shared<op::Squeeze>(t, cnst_axis_lable);
+        auto t_s_matcher = make_shared<pattern::Matcher>(s, "transpose_squeeze_pattern");
```

Comment:
You don't need to use the pattern matcher here -- just check if the inputs correct.